### PR TITLE
Inventory system

### LINKED ISSUE #21: Inventory system
## Objective
Implement a slot-based inventory system with carry weight limits that creates meaningful decisions about what to scavenge and carry back to the safehouse.

## Acceptance Criteria
- [ ] Inventory defined as a fixed-size array of item slots on the player entity (e.g., 8 slots)
- [ ] Each object occupies one or more slots based on its size category (small = 1, medium = 2, large = not carryable)
- [ ] Total carry weight tracked — exceeding a threshold progressively reduces player movement speed
- [ ] InventorySystem manages add, remove, swap, and reorder operations
- [ ] Quick-select slots bound to number keys (1-5) for active item selection
- [ ] Active item displayed as equipped (visual indicator on player sprite placeholder)
- [ ] Inventory state bridged to Zustand via event bus for UI rendering
- [ ] Attempting to pick up an item with a full inventory shows a "full" notification
- [ ] Dropping items from inventory places them in the world at the player's feet

## Technical Notes
- The inventory is a component on the player entity, not a separate store
- Item data references the central object definition table (from issue #15) for properties
- Weight calculation should be a utility function that sums item masses from the definition table
- Movement speed reduction should be a smooth curve, not a hard threshold
- The inventory UI itself is built in M5 — this issue builds the data layer and system logic

## Dependencies
- Object interaction system (pick up, drop, examine)

### PR DESCRIPTION
**Pull request by allan-mobley-jr**
## Summary
Closes #21: Implement slot-based inventory system with carry weight limits, quick-select keys, and event bus bridging for UI rendering.

## Changes
- Redesigned inventory from stacked array to 8 fixed slots (small=1 slot, medium=2 slots, large=not carryable)
- Added `SizeCategory` type and `getSizeSlots()` to object definitions; all 15 types classified
- Pure inventory utility functions: `addItem`, `removeItem`, `removeItemByType`, `swapItems`, `canAddItem`, `getActiveItem`, `recomputeCarryWeight`, `buildEventSlots`
- `InventorySystem` for quick-select processing (number keys 1-5, edge-triggered)
- Rewritten `InteractionSystem` pickup/drop flow with validate-before-remove and re-add-on-failure safety
- Input system captures number keys ONE-FIVE with edge detection
- Equipped item visual indicator (colored square near player sprite)
- Event bus events: `active-slot-changed`, `inventory-full`, updated `inventory-changed`
- Zustand bridge wires all new events to `usePlayerStore`
- Fixed 3 bugs from Temperer review: drop-by-slot forward-search, ungated pickup events, unchecked re-add

## Test Coverage
- `inventory-utils.test.ts`: 522 lines, ~70 cases covering all slot operations and edge cases
- `inventory-system.test.ts`: 164 lines, ~12 cases for quick-select processing
- `interaction-system.test.ts`: 1155 lines, ~60+ cases including 3 regression tests for fixed bugs
- `input-system.test.ts`: +129 lines — E key edge detection (4 tests) and number keys 1-5 mapping (7 tests)
- `object-defs.test.ts`: +60 lines — `getSizeSlots` (3 tests), sizeCategory data integrity (9 tests)
- `render-sync-system.test.ts`: +255 lines — equipped item indicator lifecycle (9 tests)
- `bridge.test.ts` and `usePlayerStore.test.ts`: updated for new events and state

## Acceptance Criteria
- [x] Inventory defined as a fixed-size array of item slots on the player entity (8 slots)
- [x] Each object occupies one or more slots based on its size category (small = 1, medium = 2, large = not carryable)
- [x] Total carry weight tracked — exceeding threshold progressively reduces player movement speed
- [x] InventorySystem manages add, remove, swap, and reorder operations
- [x] Quick-select slots bound to number keys (1-5) for active item selection
- [x] Active item displayed as equipped (visual indicator on player sprite placeholder)
- [x] Inventory state bridged to Zustand via event bus for UI rendering
- [x] Attempting to pick up an item with a full inventory shows a "full" notification
- [x] Dropping items from inventory places them in the world at the player's feet

## Quality Checks
- [x] Lint passes (0 errors, 8 pre-existing warnings)
- [x] Type check passes
- [x] Tests pass (1124 total)
- [x] Build succeeds

### CODE REVIEW

File: src/game/systems/inventory-utils.ts [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+  // Extract item data
+  const itemA = a !== null ? inventory.slots[a] : null;
+  const itemB = b !== null ? inventory.slots[b] : null;
+
+  const slotsA = itemA ? getSizeSlots(itemA.sizeCategory) : 0;
+  const slotsB = itemB ? getSizeSlots(itemB.sizeCategory) : 0;
+
+  // Check bounds after swap
+  if (itemA && b !== null && b + slotsA > inventory.slots.length) return false;
+  if (itemB && a !== null && a + slotsB > inventory.slots.length) return false;
```
[the remark this answers is not in the document]
Comment (allan-mobley-jr) [reply src/game/systems/inventory-utils.ts]: This is actually handled correctly — the overlap-check loop at lines 232-240 catches the out-of-bounds case because `inventory.slots[8]` returns `undefined`, and `undefined !== null` evaluates to `true`, triggering the rollback. The existing test at line 401 (`swapItems(inv, 0, 7)` with a medium item) verifies this path returns `false` and preserves the original positions. The implicit bounds check via JS semantics is correct, though an explicit bounds guard would make the intent clearer — noting as a future hardening opportunity.

File: src/game/systems/render-sync-system.ts [hunk 2]
```diff
@@ -157,6 +167,39 @@ export function createRenderSyncSystem(ctx: SceneContext): SystemFn {
           );
           aimGfx.strokePath();
         }
+
+        // --- Equipped item indicator ---
+        const invPlayer = inventoryEntities.entities[0];
+        if (invPlayer) {
+          const inv = invPlayer.inventory;
```
[the remark this answers is not in the document]
Comment (allan-mobley-jr) [reply src/game/systems/render-sync-system.ts]: Good observation, though this path isn't reachable today because only one player entity exists — both `playerEntities.entities[0]` and `inventoryEntities.entities[0]` reference the same entity. The suggested fix `(player as any).inventory` bypasses type safety which is worse. If co-op is added, the proper fix would be a query intersection or entity correlation.
